pkg/service/repair/helper_integration_test.go

```diff
@@ -0,0 +1,467 @@
+	if globalNodeInfo.EnableTablets {
+		stmt += " AND tablets = {'enabled': false}"
+	}
+	ExecStmt(t, session, fmt.Sprintf(stmt, keyspace, rf1, rf2))
+}
+
+// Creates tablet keyspace or skips the test if that's not possible.
+func createTabletKeyspace(t *testing.T, session gocqlx.Session, keyspace string, rf1, rf2 int) {
+	if !globalNodeInfo.EnableTablets {
+		t.Skip("Test requires tablets enabled in order to create tablet keyspace")
+	}
+	stmt := "CREATE KEYSPACE %s WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': %d, 'dc2': %d} AND tablets = {'enabled': true}"
+	ExecStmt(t, session, fmt.Sprintf(stmt, keyspace, rf1, rf2))
+}
+
+// Creates keyspace with default replication type (vnode or tablets).
+func createDefaultKeyspace(t *testing.T, session gocqlx.Session, keyspace string, rf1, rf2 int) {
+	if globalNodeInfo.EnableTablets {
+		stmt := "CREATE KEYSPACE %s WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': %d, 'dc2': %d} AND tablets = {'enabled': true}"
+		ExecStmt(t, session, fmt.Sprintf(stmt, keyspace, rf1, rf2))
+		return
+	}
+	stmt := "CREATE KEYSPACE %s WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': %d, 'dc2': %d}"
+	ExecStmt(t, session, fmt.Sprintf(stmt, keyspace, rf1, rf2))
+}
+
+func dropKeyspace(t *testing.T, session gocqlx.Session, keyspace string) {
+	ExecStmt(t, session, fmt.Sprintf("DROP KEYSPACE IF EXISTS %q", keyspace))
+}
+
+type repairSchedReq struct {
+	// always set
+	host       string
+	keyspace   string
+	table      string
+	replicaSet []string
+	ranges     []scyllaclient.TokenRange
+
+	// optional
+	SmallTableOptimization bool
+	RangesParallelism      int
+}
+
+func (r repairSchedReq) fullTable() string {
+	return r.keyspace + "." + r.table
+}
+
+type repairStatusReq struct {
+	host string
+	id   string
+}
+
+type repairSchedResp struct {
+	repairSchedReq
+	id string
+}
+type repairStatusResp struct {
+	repairStatusReq
+	status repairStatus
+}
+
+type repairStatus int
+
+const (
+	repairStatusDone repairStatus = iota
+	repairStatusFailed
+	repairStatusRunning
+)
+
+const (
+	oldRepairSchedPathPrefix  = "/storage_service/repair_async"
+	oldRepairStatusPathPrefix = "/storage_service/repair_status"
+	killRepairPathPrefix      = "/storage_service/force_terminate_repair"
+)
+
+func isOldRepairSchedReq(req *http.Request) bool {
+	return strings.HasPrefix(req.URL.Path, oldRepairSchedPathPrefix) && req.Method == http.MethodPost
+}
+
+func isRepairSchedReq(req *http.Request) bool {
+	return isOldRepairSchedReq(req)
+}
+
+func isOldRepairStatusReq(req *http.Request) bool {
+	return strings.HasPrefix(req.URL.Path, oldRepairStatusPathPrefix) && req.Method == http.MethodGet
+}
+
+func isRepairStatusReq(req *http.Request) bool {
+	return isOldRepairStatusReq(req)
+}
+
+func isKillRepairReq(req *http.Request) bool {
+	return strings.HasPrefix(req.URL.Path, killRepairPathPrefix) && req.Method == http.MethodPost
+}
+
+func newRepairSchedReq(t *testing.T, req *http.Request) (repairSchedReq, bool) {
```

Comment:
Q: What is the meaning of the `repairSchedReq{}, false` return ? What does it indicate ? What information it holds ?

Reply:
I will update it with the comment. In order to simplify interceptors code, this function can be used on any request. If it's used on a request scheduling a repair, it will return parsed request and true. If it's used on any other request, it will return `repairSchedReq{}, false`.